### PR TITLE
[infrastructure] test project-base with only one domain on GitHub actions

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, the reason for the PR| We strive to improve our testing process as much as possible so we want to test build of project-base for only one domain set.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://tl-single-domain-project-base.odin.shopsys.cloud
  - https://cz.tl-single-domain-project-base.odin.shopsys.cloud

### CODE REVIEW

File: project-base/.github/workflows/run-checks-tests.yaml [hunk 1]
```diff
@@ -35,3 +35,48 @@ jobs:
                 with:
                     name: acceptance-logs
                     path: ./var/log
+    build-whole-application-for-single-domain:
+        name: Build application and run standards checks and tests on single domain
+        runs-on: 'ubuntu-20.04'
```
Comment (vitek-rostislav) [inline project-base/.github/workflows/run-checks-tests.yaml]: in https://github.com/shopsys/shopsys/pull/2875, you used ubuntu 22, why not here?

File: project-base/.github/workflows/run-checks-tests.yaml [hunk 2]
```diff
@@ -35,3 +35,48 @@ jobs:
...
+                run: echo 1 | ./scripts/configure.sh
+            -   name: Remove other domains than first one from configs
+                run: |
+                    # delete all domain URLs except URLs for domain with ID 1 in app/config/domains.yaml
+                    sed -i '/id: 2/{:a;Q}' app/config/domains_urls.yaml
+                    sed -i '/id: 2/{:a;Q}' app/config/domains.yaml
+
+                    # sed is removing all content from match to end of file, unfortunately it adds new line (\n) after the replacement
+                    # more info: https://stackoverflow.com/questions/50559170/delete-last-line-break-using-sed
+                    # when there stays more than one last line the build will fail because of our coding standards
```
Comment (vitek-rostislav) [inline project-base/.github/workflows/run-checks-tests.yaml]:
```diff
- when there stays more than one last line the build will fail because of our coding standards
+ when there is more than one last line left, the build will fail because of our coding standards
```

File: project-base/.github/workflows/run-checks-tests.yaml [hunk 3]
```diff
@@ -35,3 +35,48 @@ jobs:
...
+        steps:
+            -   name: GIT checkout branch - ${{ github.ref }}
+                uses: actions/checkout@v4
+                with:
+                    ref: ${{ github.ref }}
+            -   name: Configure application
+                run: echo 1 | ./scripts/configure.sh
+            -   name: Remove other domains than first one from configs
+                run: |
+                    # delete all domain URLs except URLs for domain with ID 1 in app/config/domains.yaml
```
Comment (vitek-rostislav) [inline project-base/.github/workflows/run-checks-tests.yaml]:
```diff
- delete all domain URLs except URLs for domain with ID 1 in app/config/domains.yaml
+ delete all domain URLs except URLs for domain with ID 1 in app/config/domains.yaml and app/config/domains_urls.yaml
```

File: project-base/.github/workflows/run-checks-tests.yaml [hunk 4]
```diff
@@ -35,3 +35,48 @@ jobs:
...
+                    sed -i '/id: 2/{:a;Q}' app/config/domains.yaml
+
+                    # sed is removing all content from match to end of file, unfortunately it adds new line (\n) after the replacement
+                    # more info: https://stackoverflow.com/questions/50559170/delete-last-line-break-using-sed
+                    # when there stays more than one last line the build will fail because of our coding standards
+                    truncate -s -1 app/config/domains_urls.yaml
+                    truncate -s -1 app/config/domains.yaml
+            -   name: Install composer dependencies
+                run: docker-compose exec -T php-fpm composer install
+            -   name: Ensure Symfony 4 version
```
Comment (vitek-rostislav) [inline project-base/.github/workflows/run-checks-tests.yaml]: this is probably not necessary anymore - we use Symfony 5 already, right? Or maybe, it is necessary and the name should be changed to `Ensure Symfony 5 version`? :smile: see https://github.com/shopsys/shopsys/commit/e4dd48defd9f32a1fbd24e54d9812061f383afc1

File: project-base/.github/workflows/run-checks-tests.yaml [hunk 5]
```diff
@@ -35,3 +35,48 @@ jobs:
...
+                    # delete all domain URLs except URLs for domain with ID 1 in app/config/domains.yaml
+                    sed -i '/id: 2/{:a;Q}' app/config/domains_urls.yaml
+                    sed -i '/id: 2/{:a;Q}' app/config/domains.yaml
+
+                    # sed is removing all content from match to end of file, unfortunately it adds new line (\n) after the replacement
+                    # more info: https://stackoverflow.com/questions/50559170/delete-last-line-break-using-sed
+                    # when there stays more than one last line the build will fail because of our coding standards
+                    truncate -s -1 app/config/domains_urls.yaml
+                    truncate -s -1 app/config/domains.yaml
+            -   name: Install composer dependencies
```
Comment (vitek-rostislav) [inline project-base/.github/workflows/run-checks-tests.yaml:57]: I see there are a lot of duplicated lines, could we do something with that? Is not it possible to extract the common lines? (from "Install composer dependencies" to "Upload Codeception logs to artifacts", the steps are identical for both "build-whole-application" and "build-whole-application-for-single-domain" jobs, right?

File: project-base/app/src/DataFixtures/Demo/ParameterDataFixture.php [hunk 6]
```diff
@@ -158,8 +158,8 @@ public function load(ObjectManager $manager)
         $product2 = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '2');
 
         foreach ($this->domain->getAllLocales() as $locale) {
-            $parameterValueRed = $this->getParameterValue($locale, t('black', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $locale), '#000000');
-            $this->addParameterValueToProduct($product2, $parameterColor, $parameterValueRed);
+            $parameterValueBlack = $this->getParameterValue($locale, t('black', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $locale), '#000000');
+            $this->addParameterValueToProduct($product2, $parameterColor, $parameterValueBlack);
             $parameterValueMetal = $this->getParameterValue($locale, t('metal', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $locale));
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/ParameterDataFixture.php:163]: in the following two foreaches, there are also bad names for the variables - there is `$parameterValueMetal` while it should be `$parameterValuePlastic` and `$parameterValueWood`, could you please fix that as well? :pray:

File: project-base/app/src/DataFixtures/Demo/ReadyCategorySeoDataFixture.php [hunk 7]
```diff
@@ -54,13 +54,13 @@ public function load(ObjectManager $manager)
             'ordering' => ProductListOrderingConfig::ORDER_BY_PRIORITY,
             'parameterValueIdsByParameterIds' => [
                 // 'Water Tank Volume' => 2 l
-                38 => 75,
+                38 => $this->parameterFacade->getParameterValueByValueTextAndLocale(t('2 l', [], Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, $firstDomainLocale), $firstDomainLocale)->getId(),
```
Comment (vitek-rostislav) [inline project-base/app/src/DataFixtures/Demo/ReadyCategorySeoDataFixture.php:57]: love it! :heart_eyes: 
If we wanted to make it perfect, we could rewrite the parameters' IDs as well :innocent: However, that would require more changes (this is on my wishlist for the future) - we would need to add references for the parameter's data fixtures that are now (mostly) created in `ProductDataFxiture::findParameterByNamesOrCreateNew`, or create a separate data fixture for the parameters, or something like that :slightly_smiling_face:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: project-base/.github/workflows/run-checks-tests.yaml (hunk 1 above)
Comment (TomasLudvik) [reply project-base/.github/workflows/run-checks-tests.yaml]: I have added new commit, where I unify it across all workflows

File: project-base/.github/workflows/run-checks-tests.yaml (hunk 2 above)
Comment (TomasLudvik) [reply project-base/.github/workflows/run-checks-tests.yaml]: It was copy-paste, but fixed

File: project-base/.github/workflows/run-checks-tests.yaml (hunk 3 above)
Comment (TomasLudvik) [reply project-base/.github/workflows/run-checks-tests.yaml]: Same here

File: project-base/.github/workflows/run-checks-tests.yaml (hunk 4 above)
Comment (TomasLudvik) [reply project-base/.github/workflows/run-checks-tests.yaml]: I have removed it

File: project-base/.github/workflows/run-checks-tests.yaml (hunk 5 above)
Comment (TomasLudvik) [reply project-base/.github/workflows/run-checks-tests.yaml:57]: Well, it is easier to keep this as it is for easier readability

File: project-base/app/src/DataFixtures/Demo/ReadyCategorySeoDataFixture.php (hunk 7 above)
Comment (TomasLudvik) [reply project-base/app/src/DataFixtures/Demo/ReadyCategorySeoDataFixture.php:57]: Yes

**Review by vitek-rostislav — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**